toonz/sources/toonz/preferencespopup.cpp

```diff
@@ -826,7 +826,8 @@ void PreferencesPopup::onDefLevelTypeChanged(int index) {
   if (0 <= index && index < m_defLevelType->count()) {
     int levelType = m_defLevelType->itemData(index).toInt();
     m_pref->setDefLevelType(levelType);
-    bool isRaster = levelType != PLI_XSHLEVEL;
+    bool isRaster =
+        levelType != PLI_XSHLEVEL && !m_newLevelToCameraSizeCB->isChecked();
     m_defLevelWidth->setEnabled(isRaster);
     m_defLevelHeight->setEnabled(isRaster);
     if (!m_pixelsOnlyCB->isChecked()) m_defLevelDpi->setEnabled(isRaster);
```

Comment:
If the `New Levels Default to the Current Camera Size` option in ON, the DPI field becomes disabled as well as the width and the height fields.
But for now the created level refers the value in the DPI field, not the current camera's dpi.

I think the level should be created with the current camera dpi, or, the DPI field should remain enabled even if this option is activated.
(IMO the former way would be simple and matches the purpose of this option.)